lib/expression.ex

```diff
@@ -96,6 +98,9 @@ defmodule Expression do
     end
   end
 
+  @spec time_struct?(String.t() | Number.t() | Time.t()) :: boolean
```

Comment:
It's not possible to have a `Number` struct because it's converted to string before calling `parse/1`: https://github.com/turnhub/expression/blob/8e0ff5d6bcb69ca01bbefb8984c7e0f5e26ed35d/lib/expression.ex#L87-L87
```suggestion
  @spec time_struct?(String.t() | Time.t()) :: boolean
```